datafusion/common/src/tree_node.rs

```diff
@@ -292,6 +292,20 @@ pub trait TreeNode: Sized {
         )
     }
 
+    fn exists<F: FnMut(&Self) -> bool>(&self, mut f: F) -> bool {
```

Comment:
Can we please add some documentation to this method?

Perhaps something like

```suggestion
    /// Returns true if `f` returns true for node in the tree.
    ///
    /// Stops recursion as soon as a matching node is found
    fn exists<F: FnMut(&Self) -> bool>(&self, mut f: F) -> bool {
```